### ISSUE TITLE
registerHandler before a watchX causes a crash

### ISSUE THREAD

**Issue by formvoltron** [escamoteur/get_it_mixin#8]
I opened this one in flutter_command, but it turns out to be an issue with get_it_mixin.  It is caused by registerHandler using watchX internally.  This causes a runtime error.

 I found a situation that causes an error and it isn't clear why it's an error.

If I use a
registerHandler((StringManager sm) => sm.transformString, (context, name, _) => print(name));
before
var message2 = watchX((StringManager sm) => sm.transformString);

this causes an error. However if I reverse the order of the statements, there is no error & everything works just fine.

What's going on here?

Thank you!
```dart
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter_command/flutter_command.dart';
import 'package:get_it/get_it.dart';
import 'package:get_it_mixin/get_it_mixin.dart';

void main() {
  var sm = StringManager();
  GetIt.I.registerSingleton<StringManager>(sm);
  sm.loadStringCommand.execute();

  runApp(MaterialApp(home: MyApp()));
}

class StringManager {
  Command<void, String> loadStringCommand;
  ValueListenable<String> transformString;

  StringManager() {
    loadStringCommand =
        Command.createAsync<void, String>(_stringService, 'initial');

    transformString =
        loadStringCommand.map<String>((result) => result.toUpperCase());
  }

//TODO why is 'void x' needed as a parameter?
  Future<String> _stringService(void x) async {
    await Future.delayed(Duration(seconds: 2));
    return 'thank you for waiting';
  }
}

class MyApp extends StatelessWidget with GetItMixin {
  @override
  Widget build(BuildContext context) {


    var message = watchX((StringManager sm) => sm.loadStringCommand);
    var message2 = watchX((StringManager sm) => sm.transformString);

    registerHandler((StringManager sm) => sm.transformString,
            (context, name, _) => print(name));

    return Scaffold(
      appBar: AppBar(
        title: Text('hello'),
      ),
      body: Column(
        children: [Text(message), Text(message2)],
      ),
    );
  }
}
```

**Comment by escamoteur** [comment]
did you solve this?

**Comment by formvoltron** [comment]
Well I filed it as an documentation bug.  Though maybe this is already in
the docs?

**Comment by escamoteur** [comment]
I will have a closer look. this shouldn't be a problem at all. sorry it slipped my attention

**Comment by formvoltron** [comment]
Certainly not a problem.  I said that I would help with docs, but then life
took over.  I still mean to do it, but I don't have much time these days.

**Comment by escamoteur** [comment]
ah, I know found the problem. you didn't write what sort of runtime error you got. 
should be fixed in V3.1.2
